CONTRIBUTING.md

```diff
@@ -370,7 +370,7 @@ same as with GSoC, and ZSoC has no separate application process; your
 GSoC application is your ZSoC application. If we'd like to select you
 for ZSoC, we'll contact you when the GSoC results are announced.
 
-[gsoc-guide]: https://zulip.readthedocs.io/en/latest/contributing/gsoc-ideas.html
+[gsoc-guide]: /contributing/gsoc-ideas.md
```

Comment:
I think we use absolute links in README.md and CONTRIBUTING.md to transfer folks who land on the GitHub copy of this page to the better ReadTheDocs version.

Reply:
Oh! Thanks for the context.
I've updated the PR to keep the link as it is (instead of changing it to `.../gsoc.html`, because that gives 404s as of now (which was my motivation to change to a relative link)), assuming we'll add a redirect.